File: backend/order-service/src/main/java/com/ecommerce/orderservice/service/OrderService.java
package com.ecommerce.orderservice.service;

import com.ecommerce.orderservice.dto.CreateOrderRequest;
import com.ecommerce.orderservice.dto.OrderDTO;

import java.util.List;

/**
 * OrderService interface - defines all order-related business operations.
 */
public interface OrderService {

    /** Place a new order */
    OrderDTO createOrder(CreateOrderRequest request);

    /** Get a single order by ID */
    OrderDTO getOrderById(Long id);

    /** Get all orders for a specific user */
    List<OrderDTO> getOrdersByUser(Long userId);

    /** Get all orders (admin) */
    List<OrderDTO> getAllOrders();

    /** Update order status (e.g. PENDING -> CONFIRMED) */
    OrderDTO updateOrderStatus(Long id, String status);

    /** Cancel/delete an order */
    void deleteOrder(Long id);
}

File: frontend/src/services/orderService.js
import { orderApi } from './api'

// Create a new order
export const createOrder = (data) =>
  orderApi.post('/api/orders', data)

// Get all orders for a specific user
export const getOrdersByUser = (userId) =>
  orderApi.get(`/api/orders/user/${userId}`)

// Get a single order by ID
export const getOrderById = (id) =>
  orderApi.get(`/api/orders/${id}`)

// Get all orders (admin)
export const getAllOrders = () =>
  orderApi.get('/api/orders')

// Update the status of an order (e.g. PENDING -> CONFIRMED)
export const updateOrderStatus = (id, status) =>
  orderApi.put(`/api/orders/${id}/status?status=${status}`)
